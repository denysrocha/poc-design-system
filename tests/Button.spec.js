import { shallowMount } from '@vue/test-utils';
import PocButton from '@/components/Button.vue';

describe('PocButton.vue', () => {
  it('renders button with correct HTML', () => {
    const wrapper = shallowMount(PocButton, {
      props: { label: 'Click me', type: 'primary', size: 'medium' },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
  
  it('renders button with provided label', () => {
    const label = 'Click me';
    const wrapper = shallowMount(PocButton, {
      props: { label },
    });
    expect(wrapper.text()).toContain(label);
  });

  it('renders button with correct type class', () => {
    const type = 'secondary';
    const label = 'Click me';
    const wrapper = shallowMount(PocButton, {
      props: { type, label },
    });
    expect(wrapper.classes(`poc-button--${type}`)).toBe(true);
  });

  it('renders button with correct size class', () => {
    const size = 'large';
    const label = 'Click me';
    const wrapper = shallowMount(PocButton, {
      props: { size, label },
    });
    expect(wrapper.classes(`poc-button--${size}`)).toBe(true);
  });

  it('renders button as disabled when disabled prop is true', () => {
    const label = 'Click me';
    const wrapper = shallowMount(PocButton, {
      props: { disabled: true, label },
    });
    expect(wrapper.attributes('disabled')).toBeDefined();
  });
  

  it('emits click event when button is clicked', async () => {
    const label = 'Click me';
    const wrapper = shallowMount(PocButton, {
        props: { label },
      });
    await wrapper.trigger('click');
    expect(wrapper.emitted('click')).toBeTruthy();
  });
});
