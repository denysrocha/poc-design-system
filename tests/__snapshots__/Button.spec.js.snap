// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PocButton.vue renders button with correct HTML 1`] = `"<button type="button" class="poc-button poc-button--primary poc-button--medium">Click me</button>"`;
